components/plugin_marketplace/marketplace_modal.jsx

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import debounce from 'lodash/debounce';
+import {Tabs, Tab} from 'react-bootstrap';
+
+import FullScreenModal from 'components/widgets/modals/full_screen_modal';
+import RootPortal from 'components/root_portal';
+import QuickInput from 'components/quick_input';
+import LocalizedInput from 'components/localized_input/localized_input';
+import PluginIcon from 'components/widgets/icons/plugin_icon.jsx';
+import LoadingScreen from 'components/loading_screen.jsx';
+
+import {trackEvent} from 'actions/diagnostics_actions.jsx';
+import {t} from 'utils/i18n';
+import {localizeMessage} from 'utils/utils';
+import * as Markdown from 'utils/markdown';
+
+import MarketplaceItem from './marketplace_item';
+
+import './marketplace_modal.scss';
+
+const MarketplaceTabs = {
+    ALL_PLUGINS: 'allPlugins',
+    INSTALLED_PLUGINS: 'installed',
+};
+
+const TRACK_SEARCH_WAIT = 1000;
+
+export default class MarketplaceModal extends React.Component {
+    static propTypes = {
+        show: PropTypes.bool,
+        installedPlugins: PropTypes.array.isRequired,
+        marketplacePlugins: PropTypes.array.isRequired,
+        serverError: PropTypes.object,
+        actions: PropTypes.shape({
+            closeModal: PropTypes.func.isRequired,
+            getMarketplacePlugins: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            tabKey: MarketplaceTabs.ALL_PLUGINS,
+            loading: false,
+        };
+    }
+
+    componentWillMount() {
```

Comment:
`componentWillMount` is being deprecated, we should use the constructor instead

Reply:
Moved to `componentDidMount` as a recommended migration by React [here](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#fetching-external-data)